modules/shortcodes/instagram.php

```diff
@@ -161,8 +163,49 @@ function jetpack_instagram_handler( $matches, $atts, $url ) {
 		$url_args['hidecaption'] = 'true';
 	}
 
-	$url = esc_url_raw( add_query_arg( $url_args, 'https://api.instagram.com/oembed/' ) );
+	$use_cache     = jetpack_instagram_use_cache( $matches, $atts, $passed_url );
+	$cache_key     = 'oembed_response_body_' . md5( add_query_arg( $url_args, 'https://api.instagram.com/oembed/' ) );
+	$response_body = $use_cache
+		? wp_cache_get( $cache_key, 'instagram_embeds' )
+		: false;
 
+	if ( ! $response_body ) {
+		$response_body = jetpack_instagram_fetch_embed( $url_args );
+	}
+
+	if ( $use_cache ) {
+		wp_cache_set(
+			$cache_key,
+			$response_body,
+			'instagram_embeds',
+			HOUR_IN_SECONDS + wp_rand( 0, HOUR_IN_SECONDS )
+		);
```

Comment:
In some cases, `jetpack_instagram_fetch_embed` will not return an embed. Should we avoid caching when that happens, so one does not get to keep an error message for a little while, even after they reconnected their site to WordPress.com for example?

Reply:
Great point sir! I'll get that fixed up.